.vscode/settings.json

```diff
@@ -1,4 +1,5 @@
 {
+    "files.eol": "\n",
```

Comment:
Go files should always use Unix line endings but other files should not, in case someone what's to code on Windows. Line endings for `.go`files are already handled by the top level `.gitattributes`. https://github.com/terraform-providers/terraform-provider-azuredevops/blob/257c1c7e63f4be92bba1182e81e5f45242fd4902/.gitattributes#L4

Reply:
Hmm! Good point! I don't know nearly enough about line ending. I'll move this setting back to the dev container, since that's always linux. 

On the point of file endings, shouldn't we add to gitattributes to always use a specific file ending for certain file types? I know that git gets really confused whenever you clone into a different system and you end up with hundreds of files changed.

Reply:
The infamous, often hated story of different line endings 🥴 😁 Okay, where to start. 🤔 Normally Git is very good in handling different line endings on different operating systems (okay ... different line endings between Windows and the rest 😁 ) automatically. Internally Git stores all files as Unix line endings, but during checkout it converts it to the default line endings of the current operating system. (https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings) By this, even people who use `notepad`, could edit code (text files) that are under control of Git.

But sometimes you have to pin the line endings for a specific file or file type (like `*.go`) because for other (technical) reasons. When you're using the Go language you should be a aware that the people behind Go denied the support for the CRLF line endings on Windows for all tools that come with Go like `gofmt`. That's why I filed a PR (https://github.com/terraform-providers/terraform-provider-azuredevops/commit/406481301ffe5cf10043bafe7001307b49b0c5a3) to pin the line endings to LF for *.go files, so that I could develop using Windows and I don't always run into the issue that Git is complaining about diverted line endings I reformatted the code with `gofmt` or because the Go AddIn for VSCode was so nice to do that while saving a `*.go` file.

But to me it makes little sense to pin line endings in `.gitattributes` or in the VSCode settings because Git is handling the conversion very well and, as said, ensures that people can use their favorite text editor without any hassle.

Reply:
I think it handles them less well, when you clone on windows and move to files to linux (devcontainer), no?